src/components/BaseIcon.vue

```diff
@@ -0,0 +1,50 @@
+<template>
+  <svg
+    :class="base.icon"
+    :style="{
+      width: size,
+      height: size
+    }"
+    :role="role"
+  >
+    <use :xlink:href="`#${name}`"></use>
+    <desc v-if="alt">{{alt}}</desc>
+  </svg>
+</template>
+
+<script>
+export default {
+  name: 'BaseIcon',
+  props: {
+    name: {
+      type: String,
+      required: true
```

Comment:
It might be worth validating it to a set of icons that we can use from the svg sprite? But I think thats a _super_ nice to have.